src/Compilers/CSharp/Portable/Symbols/NativeIntegerTypeSymbol.cs

```diff
@@ -0,0 +1,67 @@
+﻿// Copyright (c) Microsoft.  All Rights Reserved.  Licensed under the Apache License, Version 2.0.  See License.txt in the project root for license information.
+
+using System;
+using System.Collections.Generic;
+using System.Collections.Immutable;
+using System.Diagnostics;
+using Roslyn.Utilities;
+
+namespace Microsoft.CodeAnalysis.CSharp.Symbols
+{
+    internal sealed class NativeIntegerTypeSymbol : WrappedNamedTypeSymbol
+    {
+        internal NativeIntegerTypeSymbol(NamedTypeSymbol underlying) : base(underlying)
+        {
+            Debug.Assert(!underlying.IsNativeInt);
+            Debug.Assert(underlying.SpecialType == SpecialType.System_IntPtr || underlying.SpecialType == SpecialType.System_UIntPtr);
+            Debug.Assert(this.Equals(underlying));
+            Debug.Assert(underlying.Equals(this));
+        }
+
+        public override ImmutableArray<TypeParameterSymbol> TypeParameters => ImmutableArray<TypeParameterSymbol>.Empty;
+
+        public override NamedTypeSymbol ConstructedFrom => _underlyingType.ConstructedFrom;
+
+        public override Symbol ContainingSymbol => _underlyingType.ContainingSymbol;
+
+        internal override ImmutableArray<TypeWithAnnotations> TypeArgumentsWithAnnotationsNoUseSiteDiagnostics => ImmutableArray<TypeWithAnnotations>.Empty;
+
+        internal override bool IsComImport => _underlyingType.IsComImport;
+
+        internal override NamedTypeSymbol BaseTypeNoUseSiteDiagnostics => _underlyingType.BaseTypeNoUseSiteDiagnostics;
+
+        public override SpecialType SpecialType => _underlyingType.SpecialType;
+
+        public override IEnumerable<string> MemberNames => throw new NotImplementedException();
+
+        public override ImmutableArray<Symbol> GetMembers() => _underlyingType.GetMembers();
```

Comment:
>_underlyingType.GetMembers() [](start = 63, length = 28)

This looks suspicious, I get members that do not give ```this``` back as their container. I think we should clone members.

Reply:
> I think we should clone members.

I guess we already "pregnant" on the public API side with ITypeSymbol.WithNullableAnnotation, but I don't believe we have similar behavior for internal symbols.

---
In reply to: [370293644](https://github.com/dotnet/roslyn/pull/40994#discussion_r370293644) [](ancestors = 370293644)